mmpose/datasets/pipelines/top_down_transform.py

```diff
@@ -307,14 +316,112 @@ def _generate_target(self, cfg, joints_3d, joints_3d_visible):
+            heatmaps[i] = cv2.GaussianBlur(heatmaps[i], kernel, 0)
+            maxi = np.amax(heatmaps[i])
+            if maxi <= 1e-8:
+                target_weight[i] = 0
+                continue
+            heatmaps[i] /= maxi / 255
+
+        return heatmaps, target_weight
+
     def __call__(self, results):
         """Generate the target heatmap."""
         joints_3d = results['joints_3d']
         joints_3d_visible = results['joints_3d_visible']
 
-        target, target_weight = self._generate_target(results['ann_info'],
-                                                      joints_3d,
-                                                      joints_3d_visible)
+        assert self.encoding in ['MSRA', 'Megvii']
+
+        if self.encoding == 'MSRA':
+            if isinstance(self.sigma, list):
+                num_sigmas = len(self.sigma)
+                cfg = results['ann_info']
+                num_joints = cfg['num_joints']
+                heatmap_size = cfg['heatmap_size']
+
+                target = np.empty(
+                    (0, num_joints, heatmap_size[1], heatmap_size[0]),
+                    dtype=np.float32)
+                target_weight = np.empty((0, num_joints, 1), dtype=np.float32)
+                for i in range(num_sigmas):
+                    target_i, target_weight_i = self._msra_generate_target(
+                        cfg, joints_3d, joints_3d_visible, self.sigma[i])
+                    target = np.concatenate(
+                        [target, np.expand_dims(target_i, axis=0)], axis=0)
+                    target_weight = np.concatenate([
+                        target_weight,
+                        np.expand_dims(target_weight_i, axis=0)
+                    ],
+                                                   axis=0)
+            else:
+                target, target_weight = self._msra_generate_target(
+                    results['ann_info'], joints_3d, joints_3d_visible,
+                    self.sigma)
+        elif self.encoding == 'Megvii':
+            if isinstance(self.kernel, list):
+                num_kernels = len(self.kernel)
+                cfg = results['ann_info']
+                num_joints = cfg['num_joints']
+                heatmap_size = cfg['heatmap_size']
+
+                target = np.empty(
+                    (0, num_joints, heatmap_size[1], heatmap_size[0]),
+                    dtype=np.float32)
+                target_weight = np.empty((0, num_joints, 1), dtype=np.float32)
+                for i in range(num_kernels):
+                    target_i, target_weight_i = self._megvii_generate_target(
+                        cfg, joints_3d, joints_3d_visible, self.kernel[i])
+                    target = np.concatenate(
+                        [target, np.expand_dims(target_i, axis=0)], axis=0)
+                    target_weight = np.concatenate([
+                        target_weight,
+                        np.expand_dims(target_weight_i, axis=0)
```

Comment:
`np.expand_dims(target_weight_i, axis=0)` is `target_weight_i[None]`

Reply:
OK. I will change it to target_weight_i[None].